_drafts/nrf-connect-with-github-actions.md

```diff
@@ -0,0 +1,859 @@
+
+So we can look at every piece of software added to that image.
+
+Since our project only needs a single toolchain, let's go ahead and create our
+own, custom Docker image!
+
+### Custom Docker Image
+
+Before we dive in, I'd like to mention a recent project that maintains several
+very nice Docker images for embedded development:
+
+<https://github.com/embeddedcontainers/ncs>
+
+Including images for building Zephyr projects and nRF-Connect SDK projects. We
+can use those images as reference as we go to create our own custom image.
+
+Fortunately, we already have all the pieces we need to construct our docker
+image, from the section above where we installed all the [dependencies
+on-the-fly](#runtime-environment-setup). We add all those pieces to a docker
+image and upload it to the GitHub Container Registry, then use the image in our
+build.
+
+Here's the Dockerfile defining the image:
+
+```dockerfile
+FROM ubuntu:22.04
+
+# set noninteractive installation
+ENV DEBIAN_FRONTEND=noninteractive
+
+# install host packages for Zephyr builds. package list is from here:
+# https://docs.zephyrproject.org/3.5.0/develop/getting_started/index.html#install-dependencies
+RUN apt-get update && apt-get install -y --no-install-recommends \
+  ccache \
+  cmake \
+  file \
+  g++-multilib \
+  gcc \
+  gcc-multilib \
+  git \
+  gperf \
+  make \
+  ninja-build \
+  python3-venv \
+  wget \
+  xz-utils
+
+# install the Zephyr toolchain only for arm-zephyr-eabi
+ARG ZEPHYR_SDK_VERSION=0.16.4
+RUN wget -q https://github.com/zephyrproject-rtos/sdk-ng/releases/download/v${ZEPHYR_SDK_VERSION}/zephyr-sdk-${ZEPHYR_SDK_VERSION}_linux-x86_64_minimal.tar.xz \
+  && mkdir -p /opt/toolchains \
+  && tar xf zephyr-sdk-${ZEPHYR_SDK_VERSION}_linux-x86_64_minimal.tar.xz -C /opt/toolchains \
+  && /opt/toolchains/zephyr-sdk-${ZEPHYR_SDK_VERSION}/setup.sh -c -t arm-zephyr-eabi \
+  && rm zephyr-sdk-${ZEPHYR_SDK_VERSION}_linux-x86_64_minimal.tar.xz
+
+# tell cmake where to find the Zephyr SDK cmake packages
+ENV CMAKE_PREFIX_PATH /opt/toolchains
+
+# create a Python virtual environment and install Python packages
+RUN python3 -m venv /venv \
+  && . /venv/bin/activate \
+  && pip install --no-cache-dir \
+    west==1.2.0
+
+# fetch our target NCS version to make west update faster when we build
+COPY west.yml /zephyr-workspace-cache/.manifest/west.yml
+
+# retrieve a lighter version of the Zephyr workspace, which is enough to
+# populate a cache for use when building. we'll need to re-generate this image
+# if the manifest changes, to keep the benefit.
+RUN . /venv/bin/activate \
+  && cd /zephyr-workspace-cache \
+  && west init -l .manifest \
+  && west update --narrow --fetch-opt=--filter=tree:0
+
+# install Python dependencies for the Zephyr workspace
+RUN . /venv/bin/activate \
+  && cd /zephyr-workspace-cache \
+  && pip install --no-cache-dir -r zephyr/scripts/requirements.txt
+```
+
+We'll build the image using this command:
+
```

Comment:
Could you add this quick note for us sad Mac users:

Note: By default, the docker will automatically select an image variant which matches your OS and architecture. Users with Apple Silicon (or another ARM64 based architecture) should build the image with AMD64 architecture by passing `--platform linux/amd64` to `docker build` to avoid issues when deploying ARM64 images to a AMD64 environment, such as those we expect to used by the GitHub runners.

Reply:
ohh great callout! thank you